components/script/dom/window.rs

```diff
@@ -386,6 +386,27 @@ impl Window {
     pub fn https_state(&self) -> HttpsState {
         self.Document().https_state()
     }
+
+    /// https://w3c.github.io/webappsec-mixed-content/#categorize-settings-object
+    pub fn prohibit_mixed_security_contexts(&self) -> bool {
+        let mut responsible_document = self.Document();
+        loop {
+            if responsible_document.https_state() != HttpsState::None {
+                return true;
+            }
+            match responsible_document.window().parent_info() {
```

Comment:
This is broken for cross-origin iframes. We will need to talk to the constellation to make this work correctly.